File: server/apps/assignments/application/use_cases/update_assignment_application_class_section_status_use_case.py
class UpdateAssignmentApplicationClassSectionStatusUseCase:

    def __init__(
        self,
        assignment_application_repository,
    ):
        self.assignment_application_repository = assignment_application_repository

    def execute(
        self,
        assignment_application_id: str,
        class_section_id: str,
        teacher_email: str,
        status: str,
    ):

        if status not in (
            "active",
            "closed",
        ):
            raise ValueError("Trạng thái phải là active hoặc closed.")

        return self.assignment_application_repository.update_class_section_status(
            assignment_application_id=(assignment_application_id),
            class_section_id=class_section_id,
            teacher_email=teacher_email,
            status=status,
        )

File: client/src/application/assignments/use_cases/update_assignment_application_class_section_status.use_case.js
class UpdateAssignmentApplicationClassSectionStatusUseCase {
  constructor(assignmentApplicationRepository) {
    this.assignmentApplicationRepository = assignmentApplicationRepository;
  }

  async execute(req, { assignmentApplicationId, classSectionId, status }) {
    if (!assignmentApplicationId) {
      throw new Error("Thiếu mã application.");
    }

    if (!classSectionId) {
      throw new Error("Thiếu mã lớp học phần.");
    }

    if (!["active", "closed"].includes(status)) {
      throw new Error("Trạng thái phải là active hoặc closed.");
    }

    return this.assignmentApplicationRepository.updateClassSectionStatus(
      req,
      assignmentApplicationId,
      classSectionId,
      status,
    );
  }
}

module.exports = UpdateAssignmentApplicationClassSectionStatusUseCase;
